File: frontend/src/presentation.js
export const createEmptyPresentation = (index) => ({
  id: null,
  title: `Презентация ${index}`,
  slides: null,
});

export const getNextPresentationTitle = (presentations) => {
  const usedTitles = new Set(presentations.map((presentation) => presentation.title));
  let index = presentations.length + 1;

  while (usedTitles.has(`Презентация ${index}`)) {
    index += 1;
  }

  return `Презентация ${index}`;
};

File: src/main/java/ru/presentation/domain/Presentation.java
package ru.presentation.domain;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OrderColumn;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "presentations")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Presentation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Builder.Default
    @OrderColumn(name = "slide_order")
    @OneToMany(mappedBy = "presentation", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Slide> slides = new ArrayList<>();

    public void replaceSlides(List<Slide> nextSlides) {
        slides.clear();
        if (nextSlides == null) {
            return;
        }

        nextSlides.forEach(this::addSlide);
    }

    public void addSlide(Slide slide) {
        slide.setPresentation(this);
        slides.add(slide);
    }
}
